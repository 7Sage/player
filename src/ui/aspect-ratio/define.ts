import { safelyDefineCustomElement } from '../../utils/dom';
import {
  ASPECT_RATIO_ELEMENT_TAG_NAME,
  AspectRatioElement
} from './AspectRatioElement';

safelyDefineCustomElement(ASPECT_RATIO_ELEMENT_TAG_NAME, AspectRatioElement);

declare global {
  interface HTMLElementTagNameMap {
    [ASPECT_RATIO_ELEMENT_TAG_NAME]: AspectRatioElement;
  }
}
