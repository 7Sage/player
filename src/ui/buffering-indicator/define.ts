import { safelyDefineCustomElement } from '@utils/dom';

import {
  BUFFERING_INDICATOR_ELEMENT_TAG_NAME,
  BufferingIndicatorElement
} from './BufferingIndicatorElement';

safelyDefineCustomElement(
  BUFFERING_INDICATOR_ELEMENT_TAG_NAME,
  BufferingIndicatorElement
);

declare global {
  interface HTMLElementTagNameMap {
    [BUFFERING_INDICATOR_ELEMENT_TAG_NAME]: BufferingIndicatorElement;
  }
}
