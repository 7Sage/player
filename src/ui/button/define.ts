import { safelyDefineCustomElement } from '../../utils/dom';
import { BUTTON_ELEMENT_TAG_NAME, ButtonElement } from './ButtonElement';

safelyDefineCustomElement(BUTTON_ELEMENT_TAG_NAME, ButtonElement);

declare global {
  interface HTMLElementTagNameMap {
    [BUTTON_ELEMENT_TAG_NAME]: ButtonElement;
  }
}
