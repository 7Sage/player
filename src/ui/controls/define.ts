import { safelyDefineCustomElement } from '@utils/dom';

import { CONTROLS_ELEMENT_TAG_NAME, ControlsElement } from './ControlsElement';

safelyDefineCustomElement(CONTROLS_ELEMENT_TAG_NAME, ControlsElement);

declare global {
  interface HTMLElementTagNameMap {
    [CONTROLS_ELEMENT_TAG_NAME]: ControlsElement;
  }
}
