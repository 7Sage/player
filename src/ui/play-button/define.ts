import { safelyDefineCustomElement } from '@utils/dom';

import {
  PLAY_BUTTON_ELEMENT_TAG_NAME,
  PlayButtonElement
} from './PlayButtonElement';

safelyDefineCustomElement(PLAY_BUTTON_ELEMENT_TAG_NAME, PlayButtonElement);

declare global {
  interface HTMLElementTagNameMap {
    [PLAY_BUTTON_ELEMENT_TAG_NAME]: PlayButtonElement;
  }
}
