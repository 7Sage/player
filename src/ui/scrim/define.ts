import { safelyDefineCustomElement } from '../../utils/dom';
import { SCRIM_ELEMENT_TAG_NAME, ScrimElement } from './ScrimElement';

safelyDefineCustomElement(SCRIM_ELEMENT_TAG_NAME, ScrimElement);

declare global {
  interface HTMLElementTagNameMap {
    [SCRIM_ELEMENT_TAG_NAME]: ScrimElement;
  }
}
