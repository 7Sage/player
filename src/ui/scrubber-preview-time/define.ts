import { safelyDefineCustomElement } from '../../utils/dom';
import {
  SCRUBBER_PREVIEW_TIME_ELEMENT_TAG_NAME,
  ScrubberPreviewTimeElement
} from './ScrubberPreviewTimeElement';

safelyDefineCustomElement(
  SCRUBBER_PREVIEW_TIME_ELEMENT_TAG_NAME,
  ScrubberPreviewTimeElement
);

declare global {
  interface HTMLElementTagNameMap {
    [SCRUBBER_PREVIEW_TIME_ELEMENT_TAG_NAME]: ScrubberPreviewTimeElement;
  }
}
