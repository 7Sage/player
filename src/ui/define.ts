import './buffering-indicator/define';
import './controls/define';
import './fullscreen-button/define';
import './mute-button/define';
import './play-button/define';
import './scrim/define';
import './scrubber/define';
import './scrubber-preview/define';
import './scrubber-preview-time/define';
import './scrubber-preview-video/define';
import './seekable-progress-bar/define';
import './slider/define';
import './time/define';
import './time-current/define';
import './time-duration/define';
import './time-slider/define';
import './toggle-button/define';
import './volume-slider/define';

export * from '.';
