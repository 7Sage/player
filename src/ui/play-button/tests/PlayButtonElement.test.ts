import '../../../define/vds-play-button';

import { elementUpdated } from '@open-wc/testing-helpers';
import { html } from 'lit';

import { waitForEvent } from '../../../global/tests/utils';
import { buildMediaPlayerFixture } from '../../../media/test-utils';

async function buildFixture() {
  const { player } = await buildMediaPlayerFixture(html`
    <vds-play-button>
      <div class="play"></div>
      <div class="pause"></div>
    </vds-play-button>
  `);

  const button = player.querySelector('vds-play-button')!;

  return { player, button };
}

test('light DOM snapshot', async function () {
  const { button } = await buildFixture();
  expect(button).dom.toMatchSnapshot();
});

test('shadow DOM snapshot', async function () {
  const { button } = await buildFixture();
  expect(button).shadowDom.to.equal(`<slot></slot>`);
});

test('it should update paused state', async function () {
  const { player, button } = await buildFixture();

  player._mediaStore.paused.set(true);
  await elementUpdated(button);

  expect(button.isPressed).to.be.false;
  expect(button.getAttribute('aria-pressed')).to.equal('false');
  expect(button.hasAttribute('media-paused')).to.be.true;

  player._mediaStore.paused.set(false);
  await elementUpdated(button);

  expect(button.isPressed).to.be.true;
  expect(button.getAttribute('aria-pressed')).to.equal('true');
  expect(button.hasAttribute('media-paused')).to.be.false;
});

test('it should play', async function () {
  const { player, button } = await buildFixture();

  const pausedSpy = vi.spyOn(player, 'paused', 'set');

  player._mediaStore.paused.set(true);
  await elementUpdated(button);

  setTimeout(() => button.click());

  await waitForEvent(button, 'vds-play-request');
  expect(pausedSpy).to.toHaveBeenCalledWith(false);
});

test('it should pause', async function () {
  const { player, button } = await buildFixture();

  const pausedSpy = vi.spyOn(player, 'paused', 'set');

  player._mediaStore.paused.set(false);
  await elementUpdated(button);

  setTimeout(() => button.click());

  await waitForEvent(button, 'vds-pause-request');
  expect(pausedSpy).to.toHaveBeenCalledWith(true);
});
