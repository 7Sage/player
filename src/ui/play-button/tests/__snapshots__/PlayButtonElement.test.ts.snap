// Vitest Snapshot v1

exports[`light DOM snapshot 1`] = `
<vds-play-button
  aria-label="Play"
  aria-pressed="false"
  media-paused=""
  role="button"
  tabindex="0"
>
  
      
  <div
    class="play"
  />
  
      
  <div
    class="pause"
  />
  
    
</vds-play-button>
`;
